cpp/tensorrt_llm/cortex.tensorrt-llm/src/tensorrt-llm_engine.cc

```diff
@@ -236,58 +275,73 @@ void TensorrtllmEngine::HandleChatCompletion(std::shared_ptr<Json::Value> json_b
 
   // tokens for Mistral v0.3
   // TODO(sang): too much hard code here, need to refactor it soon
-  std::vector<int32_t> tokens = {static_cast<int32_t>(MistralTemplate::kBos)};
-
+  std::vector<int32_t> tokens;
+  if (model_type_ == ModelType::llama3){
+    tokens.push_back(128000); // <|begin_of_text|>
+  }
```

Comment:
else branch for mistral?

Reply:
This logic is only for llama3 chat format. I followed this function https://github.com/meta-llama/llama3/blob/main/llama/tokenizer.py#L222 , they always put "begin_of_text" to the head of tokens, then loop over the list of messages and decode them. 
The mistral doesn't have that logic, it encodes every messages individually.

Reply:
```
std::vector<int32_t> tokens = {static_cast<int32_t>(MistralTemplate::kBos)};
```
The previous version initialized tokens vector with kBos for Mistral, we still need to keep that logic